src/purify.ts

```diff
@@ -1645,15 +1645,21 @@ function createDOMPurify(window: WindowLike = getGlobal()): DOMPurify {
     return _isValidAttribute(lcTag, lcName, value);
   };
 
-  DOMPurify.addHook = function (entryPoint, hookFunction) {
+  DOMPurify.addHook = function (
+    entryPoint: keyof HooksMap,
+    hookFunction: ArrayElement<HooksMap[keyof HooksMap]>
+  ) {
```

Comment:
<img width="578" height="195" alt="스크린샷 2025-07-27 오후 5 36 25" src="https://github.com/user-attachments/assets/b146e59e-b6d9-4bbb-ab41-f1e54051cf1c" />

It is better for the developer experience to have clear type inference rather than type inference with `any`.

```ts
DOMPurify.addHook(entryPoint: BasicHookName, hookFunction: NodeHook): void (+4 overloads)
```
Of course, `DOMPurify.addHook` has a well-defined type, but I am suggesting this in the actual implementation code area.

Reply:
`DOMPurify.removeHook/DOMPurify.removeHooks` are both the same proposal.